contracts/CrowdsaleToken.sol

```diff
@@ -48,6 +48,9 @@ contract CrowdsaleToken is ReleasableToken, MintableToken, UpgradeableToken {
   function CrowdsaleToken(string _name, string _symbol, uint _initialSupply, uint _decimals, bool _mintable)
     UpgradeableToken(msg.sender) {
 
+    // Cannot create a token without supply and no minting
+    require(_mintable || _initialSupply != 0);
```

Comment:
I think this should probably be moved to the same place the "throw" used to be? Minting should be optional, right?

Reply:
The require() ensures that token is either mintable or is pre-allocated an initial supply so minting is  optional as long as initial supply of tokens has been pre-allocated. Hence, I think start of function is an appropriate place for the require().